pkg/operator/resourcesynccontroller/core.go

```diff
@@ -18,7 +19,8 @@ func CombineCABundleConfigMaps(destinationConfigMap ResourceLocation, lister cor
 	certificates := []*x509.Certificate{}
 	for _, input := range inputConfigMaps {
 		inputConfigMap, err := lister.ConfigMaps(input.Namespace).Get(input.Name)
-		if apierrors.IsNotFound(err) {
+		if apierrors.IsNotFound(err) && !input.Required {
```

Comment:
comment here so it's obvious that the fall through to if err != nil is obvious

Reply:
I've restructured the code so it's easier to read and doesn't need the comment